main/src/ca/uwaterloo/flix/runtime/interpreter/Interpreter.scala

```diff
@@ -208,6 +207,40 @@ object Interpreter {
         throw InternalRuntimeException(s"Array index out of bounds: $i. Array length: ${b.elms.length}.")
       }
 
+    //
+    // NewChannel expressions.
+    //
+    case Expression.NewChannel(len, tpe, loc) =>
+      val l: Int = cast2int32(eval(len, env0, henv0, lenv0, root))
+      Value.Channel(l, tpe)
+
+    //
+    // GetChannel expressions.
+    //
+    case Expression.GetChannel(exp, tpe, loc) =>
+      val c = cast2channel(eval(exp, env0, henv0, lenv0, root))
+      c.get()
```

Comment:
The semantics should be here, not inside Value.

The Flix compiler is functional, not object-oriented. We try not to have any methods on objects.